packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -682,19 +683,45 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     return linkMarks;
   }
 
+  isValidHref(url: string): boolean {
+    if (!this._autoLinkRegexNonGlobal?.test(url)) {
+      return false;
+    }
+
+    const href = this.buildHref(url);
+
+    if (!this.isValidTLD(href)) {
+      return false;
+    }
+
+    return true;
+  }
+
   private findAutoLinks(str: string): FoundAutoLink[] {
     const toAutoLink: FoundAutoLink[] = [];
 
-    for (const match of findMatches(str, this.options.autoLinkRegex)) {
-      const text = getMatchString(match);
+    const { autoLinkAfter, autoLinkRegex } = this.options;
+
+    let matcher = autoLinkRegex;
+
+    if (autoLinkAfter) {
+      matcher = new RegExp(autoLinkRegex.source + autoLinkAfter.source, 'gm');
+    }
+
+    for (const match of findMatches(str, matcher)) {
+      let text = getMatchString(match);
+
+      if (autoLinkAfter) {
+        text = text.slice(0, -1);
```

Comment:
Might be useful to support `autoLinkAfter` matching more than one character—possible someone might assume that's possible. Not sure the best way to remove the `autoLinkAfter` character(s) in the most reliable way though.